com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -168,6 +167,14 @@ public abstract class Agent : MonoBehaviour
         /// </summary>
         DemonstrationRecorder m_Recorder;
 
+        /// <summary>
+        /// Set of DemonstrationStores that the Agent will write its step information to.
+        /// If you use a DemonstrationRecorder component, this will automatically register its DemonstrationStore.
+        /// You can also add your own DemonstrationStores; the Agent is not responsible for creating or closing the
+        /// stores, only opening them.
+        /// </summary>
+        public ISet<DemonstrationStore> DemonstrationStores = new HashSet<DemonstrationStore>();
```

Comment:
I can hide this a bit and add AddDemonstrationStore/RemoveDemonstrationStore methods.

Reply:
seems like it would be great if we could just completely remove this demonstration stuff from Agent.  It feels like it could happen.  But there is no way to hook into Agent at the moment to know when SendInfo is called on a specific one.

Reply:
We could put this in the policy factory maybe? This way we can wrap the policy in a recording policy that looks for demo recorder components at initialization.